### PR TITLE
Task/multientity plugin ngsi2

### PR DESCRIPTION
**Pull request by dcalvoalonso**
This PR provides support for multientity plugin using NGSIv2. It includes also unit tests for all modified files.
Unit tests and lint are passed.

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -5,3 +5,5 @@
 - Add: NGSIv2 support for Timestamp Compression plugin, through configuration (#527)
 - Add: NGSIv2 support for Timestamp Processing plugin, through configuration (#527)
 - Add: NGSIv2 support for Expression Translation, Attribute Alias and Event plugins, through configuration (#527)
+- Add: NGSIv2 support for Bidirectionality plugin (#527)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: It seems task/multientityPluginNgsi2 (this PR) was branched from task/bidirectionalityPlugin (PR #4), so this PR mixes changes in both case.

After merging PR #4 and merging task/activeAttsNgsi2 into task/multientityPluginNgsi2, this PR will include only the changes regarding multientity plugin and would be reviewable.

Comment (dcalvoalonso) [reply CHANGES_NEXT_RELEASE]: PR #4 has been merged in task/activeAttsNgsi2 and in task/multientityPluginNgsi2. This PR is ready to be reviewed!

File: test/unit/ngsiv2/plugins/multientity-plugin_test.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                    {
+                        object_id: 'p',
+                        name: 'pressure',
+                        type: 'Hgmm'
+                    },
+                    {
+                        object_id: 'h',
+                        name: 'humidity',
+                        type: 'Percentage',
+                        entity_name: 'Higro2000',
```
Comment (fgalan) [inline test/unit/ngsiv2/plugins/multientity-plugin_test.js:59]: entity_name and entity_type are part of the active array of attributes? The entity is associated to the device, isn't it?

(Similar comment for WeatherStation2 and WeatherStation3)

File: test/unit/ngsiv2/examples/contextRequests/updateContextMultientityPlugin2.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      "humidity": {
+        "type": "Percentage",
+        "value": "12"
+      },
+      "type": "WeatherStation",
+      "id": "Higro2000"
+    }
+  ]
+}
+
```
Comment (fgalan) [inline test/unit/ngsiv2/examples/contextRequests/updateContextMultientityPlugin2.json:22]: This file seems identical to updateContextMultientityPlugin1.json or at least I'm unable to spot the difference...

File: test/unit/ngsiv2/plugins/multientity-plugin_test.js (hunk 2 above)
Comment (fgalan) [reply test/unit/ngsiv2/plugins/multientity-plugin_test.js:59]: Maybe this refers to the entity to which that attribute maps?

File: test/unit/ngsiv2/examples/contextRequests/updateContextMultientityPlugin2.json (hunk 3 above)
Comment (fgalan) [reply test/unit/ngsiv2/examples/contextRequests/updateContextMultientityPlugin2.json:22]: I finally find the diference: entity type (here is "WeatherStation" and there is "Higrometer").

NTC. Sorry for the noise ;)

File: test/unit/ngsiv2/plugins/multientity-plugin_test.js (hunk 2 above)
Comment (dcalvoalonso) [reply test/unit/ngsiv2/plugins/multientity-plugin_test.js:59]: Yes, this is the main functionality of the MultiEntity plugin. From https://github.com/telefonicaid/iotagent-node-lib#datamapping:

> Multientity plugin (multiEntity)
> Allows the devices provisioned in the IoTAgent to map their attributes to more than one entity, declaring the target  entity through the Configuration or Device provisioning APIs.

Comment (fgalan) [reply test/unit/ngsiv2/plugins/multientity-plugin_test.js:59]: NTC

**Comment by fgalan** [comment]
LGTM. You can wait yet some time (tomorrow?) in case feedback from @AlvaroVega also comes before squash&merge the PR.

**Comment by dcalvoalonso** [comment]
Sure, no problem at all. Thanks!!

**Comment by dcalvoalonso** [comment]
It seems that there is no additional feedback from @AlvaroVega. I will squash&merge the PR.

Thanks, @fgalan  for your comments.

**Pull request merged**